### PR TITLE
Attach correct Content-Type to HTTP header

### PR DESCRIPTION
**Pull request by KSXGitHub**
Fix https://github.com/denoland/deno/issues/4553.

## Test

I did not add any test. My attempt to use `deno test` to run `file_server_test.ts` has not been successful.

### CODE REVIEW

File: tmp.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { contentType } from 'https://deno.land/std/media_types/mod.ts'
```
Comment (ry) [inline tmp.ts:1]: Remove this file

Comment (KSXGitHub) [reply tmp.ts]: @ry Done.

**Comment by KSXGitHub** [comment]
@ry

> Thanks - no need to add a test.

Does this mean that someone else will add it for me? Tests are useful against unwanted regression.

**Comment by ry** [comment]
`cargo test std_test` to run the tests.

**Comment by KSXGitHub** [comment]
@ry I am trying to compile Deno (so far, without success).

**Comment by KSXGitHub** [comment]
@ry The time it takes to clone all git submodules is too long. I guess you could just merge this then?

File: std/http/file_server.ts [hunk 2]
```diff
@@ -103,7 +104,10 @@ async function serveFile(
   const [file, fileInfo] = await Promise.all([open(filePath), stat(filePath)]);
   const headers = new Headers();
   headers.set("content-length", fileInfo.size.toString());
-  headers.set("content-type", "text/plain; charset=utf-8");
```
Comment (zhmushan) [inline std/http/file_server.ts:106]: In fact, this line of code was added by me. You know, we need to browse the `typescript` file frequently. Unfortunately, `.ts` is the extension of the video by default.

Comment (zhmushan) [reply std/http/file_server.ts:106]: You can see https://github.com/denoland/deno/issues/4056, there are some related discussions

Comment (KSXGitHub) [reply std/http/file_server.ts:106]: We can make exception for `.ts` and `.js` as a workaround.
* `.ts` → `text/plain`
* `.js` → `text/javascript`

Comment (zhmushan) [reply std/http/file_server.ts:106]: waiting for @ry 's suggestion

Comment (ry) [reply std/http/file_server.ts:106]: @zhmushan Yes I remember when we landed that - but I've changed my mind now, I think it's more useful for file_server to act as a real file server.

Comment (ry) [reply std/http/file_server.ts:106]: We should change the media_types so that .ts properly maps to typescript.

**Comment by ry** [comment]
@KSXGitHub I added a test - please take a look.

File: std/http/file_server.ts [hunk 3]
```diff
@@ -7,7 +7,8 @@
 // https://github.com/indexzero/http-server/blob/master/test/http-server-test.js
 
 const { args, stat, readdir, open, exit } = Deno;
-import { posix } from "../path/mod.ts";
+import { contentType } from "../media_types/mod.ts";
+import { posix, basename } from "../path/mod.ts";
```
Comment (zhmushan) [inline std/http/file_server.ts:11]: extname?

**Comment by KSXGitHub** [comment]
@ry I tried using deno to run the script directly:

<pre>  (zsh)   <font color="#4D9A05">deno</font> run --allow-all --reload=https://raw.githubusercontent.com/KSXGitHub/deno/file-server-content-type/std/http/file_server.ts https://raw.githubusercontent.com/KSXGitHub/deno/file-server-content-type/std/http/file_server.ts
<font color="#89E234">Download</font> https://raw.githubusercontent.com/KSXGitHub/deno/file-server-content-type/std/http/file_server.ts
<font color="#89E234">Compile</font> https://raw.githubusercontent.com/KSXGitHub/deno/file-server-content-type/std/http/file_server.ts
<font color="#EF2828"><b>error</b></font><b> TS2355</b>: A function whose declared type is neither &apos;void&apos; nor &apos;any&apos; must return a value.

► <font color="#34E2E2">https://raw.githubusercontent.com/KSXGitHub/deno/file-server-content-type/std/http/file_server.ts</font>:<font color="#FBE84F">301</font>:<font color="#FBE84F">18</font>

<span style="background-color:#D3D6CF"><font color="#000000">301</font></span> function main(): Promise&lt;void&gt; {
<span style="background-color:#D3D6CF"><font color="#000000">   </font></span> <font color="#CC0000">                 ~~~~~~~~~~~~~</font>
</pre>

screenshot

**Comment by KSXGitHub** [comment]
@ry It works great! 👍

**Pull request merged**